src/lib.rs

```diff
@@ -314,7 +321,12 @@ impl<'a> DynamicReload<'a> {
     fn should_reload(reload_path: &PathBuf, lib: &Lib) -> bool {
         if let Some(p) = lib.original_path.as_ref() {
             if reload_path.to_str().unwrap().contains(p.to_str().unwrap()) {
-                return true;
+                if cfg!(windows) {
+                    return reload_path.extension() == Some(OsStr::new("dll"));
```

Comment:
Is this needed? Shared libraries on Windows isn't required to use `.dll` to work (I know some problems calls their plugins `.p` and they are regular dlls and I wouldn't want to impose this restriction here.

Reply:
Hmm. I don't think it's needed. In my case, the library tried to reload `.dll.lib` and it crashed my program. I'll remove it from the PR as it's not related to the issue anyway.

Reply:
@gabdube My PR #15 should fix that issue with trying to load a `.dll.lib` file.  I changed the logic to check that the file extension matches.